bokehjs/src/coffee/models/formatters/datetime_tick_formatter.coffee

```diff
@@ -77,26 +61,26 @@ class DatetimeTickFormatter extends TickFormatter.Model
     'hours':        ['%Hh', '%H:%M']
     'days':         ['%m/%d', '%a%d']
     'months':       ['%m/%Y', '%b%y']
-    # TODO: the functions here currently aren't serializable
-    # and so can't be set from the python client. Although,
-    # I'm not at all sure what purpose these functions have.
-    'years':        ['%Y', _two_digit_year, _four_digit_year]
+    'years':        ['%Y']
   }
 
   # Whether or not to strip the leading zeros on tick labels.
   strip_leading_zeros: true
 
   initialize: (attrs, options) ->
     super(attrs, options)
+    @formats = _.extend({}, @_formats, @get("formats"))
```

Comment:
What's actually different about this case is the the user's input is not expected it be *all* formats. Rather, the users inputs are assumed to be just a delta that is overlaid on top of the base defaults. I think it might currently work to just set the defaults in python as is done with tool overlays now. I will try this first.